### ISSUE TITLE
Support JSPM package loading.

### ISSUE THREAD

**Issue by lostinplace** [mapbox/mapbox.js#1003]
As it stands using JSPM to load mapbox fails badly.  This is unfortunate for me, since I would like to use mapbox instead of vanilla leaflet (which works under JSPM)

I might be able to figure out how to fi this, but if you guys have ideas (or just suggestions to speed me up)  that would be really helpful

-Chris

**Comment by tmcw** [comment]
Can you be more specific about what "fails badly" means, exactly, with error messages or results?

JSPM promises

> Load any module format (ES6, AMD, CommonJS and globals) directly from any registry such as npm and GitHub with flat versioned dependency management.

So it's likely that any bug here is on JSPM's side or a usage issue.

**Comment by lostinplace** [comment]
Hey there tmcw

so I've installed mapbox using:

```
jspm install github:mapbox/mapbox.js
```

I have a javascript file that imports it using the following

```
import L from 'mapbox';
```

this yields the following errors:

```
Navigated to http://localhost:3800/map.html
2015-05-15 15:30:14.771 es6-module-loader.src.js:2612 GET http://localhost:3800/jspm_packages/github/mapbox/mapbox.js@2.1.9/package.json.js 404 (Not Found)f @ es6-module-loader.src.js:2612$__Object$defineProperty.value @ es6-module-loader.src.js:2787c @ es6-module-loader.src.js:377b @ es6-module-loader.src.js:365$__Object$defineProperty.value @ es6-module-loader.src.js:2786$__global.upgradeSystemLoader.e.fetch @ system.src.js:739$__global.upgradeSystemLoader.e.fetch @ system.src.js:1806$__global.upgradeSystemLoader.e.fetch @ system.src.js:1921(anonymous function) @ es6-module-loader.src.js:1525O @ es6-module-loader.src.js:1183K @ es6-module-loader.src.js:11427.y.when @ es6-module-loader.src.js:9307.v.run @ es6-module-loader.src.js:8213.a._drain @ es6-module-loader.src.js:973.a.drain @ es6-module-loader.src.js:62b @ es6-module-loader.src.js:268
2015-05-15 15:30:14.773 es6-module-loader.src.js:2612 GET http://localhost:3800/sanitize-caja.js 404 (Not Found)f @ es6-module-loader.src.js:2612$__Object$defineProperty.value @ es6-module-loader.src.js:2787c @ es6-module-loader.src.js:377b @ es6-module-loader.src.js:365$__Object$defineProperty.value @ es6-module-loader.src.js:2786$__global.upgradeSystemLoader.e.fetch @ system.src.js:739$__global.upgradeSystemLoader.e.fetch @ system.src.js:1806$__global.upgradeSystemLoader.e.fetch @ system.src.js:1921(anonymous function) @ es6-module-loader.src.js:1525O @ es6-module-loader.src.js:1183K @ es6-module-loader.src.js:11427.y.when @ es6-module-loader.src.js:9307.v.run @ es6-module-loader.src.js:8213.a._drain @ es6-module-loader.src.js:973.a.drain @ es6-module-loader.src.js:62b @ es6-module-loader.src.js:268
2015-05-15 15:30:14.781 es6-module-loader.src.js:2612 GET http://localhost:3800/mustache.js 404 (Not Found)f @ es6-module-loader.src.js:2612$__Object$defineProperty.value @ es6-module-loader.src.js:2787c @ es6-module-loader.src.js:377b @ es6-module-loader.src.js:365$__Object$defineProperty.value @ es6-module-loader.src.js:2786$__global.upgradeSystemLoader.e.fetch @ system.src.js:739$__global.upgradeSystemLoader.e.fetch @ system.src.js:1806$__global.upgradeSystemLoader.e.fetch @ system.src.js:1921(anonymous function) @ es6-module-loader.src.js:1525O @ es6-module-loader.src.js:1183K @ es6-module-loader.src.js:11427.y.when @ es6-module-loader.src.js:9307.v.run @ es6-module-loader.src.js:8213.a._drain @ es6-module-loader.src.js:973.a.drain @ es6-module-loader.src.js:62b @ es6-module-loader.src.js:268
2015-05-15 15:30:14.781 es6-module-loader.src.js:140 Potentially unhandled rejection [3] Error loading "github:mapbox/mapbox.js@2.1.9/package.json" at http://localhost:3800/jspm_packages/github/mapbox/mapbox.js@2.1.9/package.json.js
Error loading "github:mapbox/mapbox.js@2.1.9/package.json" from "lib/main" at http://localhost:3800/lib/main.js
Not Found: http://localhost:3800/jspm_packages/github/mapbox/mapbox.js@2.1.9/package.json.js (WARNING: non-Error used)(anonymous function) @ es6-module-loader.src.js:140f @ es6-module-loader.src.js:167i @ es6-module-loader.src.js:189
2015-05-15 15:30:14.827 es6-module-loader.src.js:2612 GET http://localhost:3800/sanitize-caja.js 404 (Not Found)f @ es6-module-loader.src.js:2612$__Object$defineProperty.value @ es6-module-loader.src.js:2787c @ es6-module-loader.src.js:377b @ es6-module-loader.src.js:365$__Object$defineProperty.value @ es6-module-loader.src.js:2786$__global.upgradeSystemLoader.e.fetch @ system.src.js:739$__global.upgradeSystemLoader.e.fetch @ system.src.js:1806$__global.upgradeSystemLoader.e.fetch @ system.src.js:1921(anonymous function) @ es6-module-loader.src.js:1525O @ es6-module-loader.src.js:1183K @ es6-module-loader.src.js:11427.y.when @ es6-module-loader.src.js:9307.v.run @ es6-module-loader.src.js:8213.a._drain @ es6-module-loader.src.js:973.a.drain @ es6-module-loader.src.js:62b @ es6-module-loader.src.js:268
2015-05-15 15:30:14.828 es6-module-loader.src.js:2612 GET http://localhost:3800/jspm_packages/github/mapbox/mapbox.js@2.1.9/package.json.js 404 (Not Found)f @ es6-module-loader.src.js:2612$__Object$defineProperty.value @ es6-module-loader.src.js:2787c @ es6-module-loader.src.js:377b @ es6-module-loader.src.js:365$__Object$defineProperty.value @ es6-module-loader.src.js:2786$__global.upgradeSystemLoader.e.fetch @ system.src.js:739$__global.upgradeSystemLoader.e.fetch @ system.src.js:1806$__global.upgradeSystemLoader.e.fetch @ system.src.js:1921(anonymous function) @ es6-module-loader.src.js:1525O @ es6-module-loader.src.js:1183K @ es6-module-loader.src.js:11427.y.when @ es6-module-loader.src.js:9307.v.run @ es6-module-loader.src.js:8213.a._drain @ es6-module-loader.src.js:973.a.drain @ es6-module-loader.src.js:62b @ es6-module-loader.src.js:268
2015-05-15 15:30:14.829 es6-module-loader.src.js:2612 GET http://localhost:3800/corslite.js 404 (Not Found)
```

the short version is this:

number 1:  it's looking for package.json.js (which doesn't exist, I can fix this by simply renaming package.json in mapbox to package.json.js), presumably because of a difference in the package manager protocol

number 2:  it's trying to load all dependencies from the root path, rather than using the jspm dependency resolver urls.  All of the libraries it requests form localhost/\*  are available, however they're in localhost/jspm_packages  and there isn't an easy way to get the library to look there without manually changing the mapbox repo.

I was just dropping an issue in place so that people come here know that the issue is a thing,  and they don't pull their hair out trying to figure out what they're doing wrong.

By hacking my application a few ways, I've got the thing to work, but it would be nice to be able to use the library without the hacks.

As it stands, it looks like mapbox-gl might integrate a little more nicely, so I'm trying that direction.

**Comment by tmcw** [comment]
Can you try installing from npm, like `jspm install npm:mapbox.js`

**Comment by lostinplace** [comment]
I had actually started from there, however 

```
import L from 'mapbox.js'
```

where mapbox.js is npm:mapbox.js  yields an L that is an empty object

**Comment by lostinplace** [comment]
It's worth noting that mapbox.js after import does correctly result in the creation of a window.L  with mapbox plugin,  but the export is not Leaflet

**Comment by lostinplace** [comment]
It occurs to me now that the simplest way to support this would be to simply add

```
module.exports = L
```

in leaflet.js, mapbox.js and index.js of npm:mapbox.js

**Comment by lostinplace** [comment]
https://github.com/mapbox/mapbox.js/pull/1004

**Comment by lostinplace** [comment]
I suspect it would also be resolved by
https://github.com/mapbox/mapbox.js/pull/725

**Comment by lostinplace** [comment]
And now #1004 even passes CI

**Comment by lostinplace** [comment]
poke

**Comment by lostinplace** [comment]
poke?

**Comment by tmcw** [comment]
Hey Chris - sorry, when I get a free moment I'll try this out, but I don't want to merge & publish without confirming that mapbox.js will continue to work standalone, bundled, with browserify, jspm, and with es6 - the potential for regressions is high.

**Comment by lostinplace** [comment]
Okie doke, I understand

**Comment by lennerd** [comment]
:+1: would love to see this working.

**Comment by mttmccb** [comment]
As would I, I'll try the hack in the short term...

**Comment by wpf500** [comment]
I would also love to see this working!

**Comment by tmcw** [comment]
Fixed in #1008
